### PR TITLE
added g:asyncomplete_min_length

### PR DESCRIPTION
**Pull request by kyouryuukunn**
I added g:asyncomplete_min_length.
Popup menu  is shown if matched length is longer it.

### CODE REVIEW

File: autoload/asyncomplete.vim [hunk 1]
```diff
@@ -276,7 +276,7 @@ function! s:remote_refresh(ctx, force) abort
         call asyncomplete#log('core', 's:remote_refresh', l:name, l:matchpos, a:ctx)
 
         let l:typed_len = l:endpos - l:startpos
-        if l:typed_len == 1
+        if !pumvisible() && l:typed_len >= g:asyncomplete_min_length
```
Comment (jsit) [inline autoload/asyncomplete.vim]: 1) The `pumvisible()` check seems a separate issue from this PR
2) I _think_ `==` should work in this context, don't need `>=`.

File: plugin/asyncomplete.vim [hunk 2]
```diff
@@ -14,9 +14,10 @@ endif
 
 let g:asyncomplete_auto_popup = get(g:, 'asyncomplete_auto_popup', 1)
 let g:asyncomplete_completion_delay = get(g:, 'asyncomplete_completion_delay', 100)
-let g:asyncomplete_default_refresh_pattern = get(g:, 'asyncomplete_default_refresh_pattern', '\(\k\+$\|\.$\|>$\|:$\)')
+let g:asyncomplete_default_refresh_pattern = get(g:, 'asyncomplete_default_refresh_pattern', '\(\k\+$\|\k\+\.\k*$\|\k\+::\?\k*$\)')
```
Comment (jsit) [inline plugin/asyncomplete.vim]: Changing the default refresh pattern feels outside of the scope of this PR

**Comment by prabirshrestha** [comment]
I'm holding off on this PR as min length PR should sent to v2 branch.

**Comment by prabirshrestha** [comment]
Closing in favor of https://github.com/prabirshrestha/asyncomplete.vim/pull/122
